jina/drivers/reduce.py

```diff
@@ -113,42 +50,31 @@ def _apply_all(
             docs: Iterable['jina_pb2.Document'],
             context_doc: 'jina_pb2.Document',
             field: str,
-            doc_pointers: Dict,
             *args,
             **kwargs) -> None:
-        if context_doc.id not in doc_pointers:
-            doc_pointers[context_doc.id] = context_doc.evaluations
+        if context_doc.id not in self.doc_pointers:
+            self.doc_pointers[context_doc.id] = context_doc.evaluations
         else:
-            doc_pointers[context_doc.id].extend(context_doc.evaluations)
+            self.doc_pointers[context_doc.id].extend(context_doc.evaluations)
 
 
-class ConcatEmbedDriver(ReduceDriver):
+class ConcatEmbedDriver(ReduceAllDriver):
```

Comment:
I think `ConcatEmbedDriver` should either disappear and changed by a `ConcatEncoder` which would be a `MultiModalEncoder` or at least separate it from the `ReduceAllDriver` (as u say the logic is orthogonal), one may want to concat the `embedding` of 2 chunks?

Reply:
If moved to a `MultiModalEncoder` then it would need to be combined with a `uses_before` = reduce

Reply:
make sense, but right now in the core I can only find `ConcatEncoder` in test: `tests/integration/multimodal/yaml/embedding-multimodal-concatenate.py`? what's the plan here?

Reply:
We added this just for testing, we could add as a real executor in the `hub`, it can be interesting.

Reply:
Actually we were desigining `MultiModalDriver` the same way `ConcatEmbedDriver` is done, but then we realised we needed to separate these concerns.

Reply:
ConcatEmbed is one level, group by `doc.id` only, `MultiModalDriver` group by `doc.id` and `doc.modality` if I understand correctly

Reply:
yes, it has some slight differences right

Reply:
no, they are irrelevant. `ConcatEmbedDriver` is a `ReduceDriver` which works on multiple requests. Complete different task than `MultiModalDriver`

Reply:
But it could be able to concatenate chunks coming from the same segmenter without the need of reducing.

Reply:
Anyway no need to change now